x/mongo/driver/operation.go

```diff
@@ -270,9 +270,9 @@ func (op Operation) getServerAndConnection(ctx context.Context) (Server, Connect
 		return nil, nil, err
 	}
 
-	// If the provided client session has a pinned connection, it should be used for the operation because this
+	// If the provided client session has a non-nil pinned connection, it should be used for the operation because this
 	// indicates that we're in a transaction and the target server is behind a load balancer.
-	if op.Client != nil && op.Client.PinnedConnection != nil {
+	if op.Client != nil && !op.Client.PinnedConnection.IsNil() {
```

Comment:
I'm not sure the issue is a "nil interface" one. What I'm trying to avoid is returning a value like `&topology.Connection{}` which isn't nil at all. Its underlying, unnamed `*connection` field is, though. Could be misunderstanding, but I don't think `reflect.ValueOf(op.Client.PinnedConnection).IsNil()` would sense that.